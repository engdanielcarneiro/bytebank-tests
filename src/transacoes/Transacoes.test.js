import { render } from "@testing-library/react";
import React from "react";
import Transacoes from "./Transacoes";

describe("Componente de listagem de transações", () => {
  it("O snapshot do componente deve sempre ser o mesmo", () => {
    const transacoes = [
      {
        id: 1,
        transacao: "saque",
        valor: "20.00",
        data: "18/03/2022",
      },
      {
        id: 2,
        transacao: "deposito",
        valor: "30.00",
        data: "15/03/2022",
      },
    ];
    const { container } = render(<Transacoes transacoes={transacoes} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
