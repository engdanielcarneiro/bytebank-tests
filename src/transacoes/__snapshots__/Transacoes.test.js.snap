// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de listagem de transações O snapshot do componente deve sempre ser o mesmo 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      18/03/2022
    </p>
    <p>
      saque
    </p>
    <p>
      R$ 
      20.00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      15/03/2022
    </p>
    <p>
      deposito
    </p>
    <p>
      R$ 
      30.00
    </p>
  </div>
</div>
`;
